pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -39,6 +39,9 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 		return err
 	}
 
+	if kc.RootDir == "" {
+		allErrors = append(allErrors, fmt.Errorf("invalid configuration: RootDir (--root-dir) must be non-empty"))
+	}
```

Comment:
what about certDir/volumePluginDir?

Reply:
Hmm. I guess adding that validation wouldn't break anyone unless they were explicitly setting those flags to the empty string. Is there any reason anyone would do that?

Reply:
(the RootDir validation was preexisting, prior in pkg/kubelet/kubelet.go)

Reply:
if they set them to the empty string, does the new defaulting kick in?

Reply:
If someone for some reason decided to _explicitly_ set them to the empty string via a _flag_, I think it would override the default, whether the default was applied in the old way or the new way.

Reply:
But yes, if they set it to the empty string in the config file, or omit it from the config file, or just use flags, the new defaulting kicks in.